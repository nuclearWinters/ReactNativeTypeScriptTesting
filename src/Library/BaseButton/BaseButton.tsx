import React from 'react';
import {
  TouchableOpacity,
  Text,
  GestureResponderEvent,
  StyleProp,
  ViewStyle,
  TextStyle,
} from 'react-native';

export interface Props {
  styles?: {
    boxStyle?: StyleProp<ViewStyle>;
    textStyle?: StyleProp<TextStyle>;
  };
  onPress(event: GestureResponderEvent): void;
  text: string;
}

const BaseButton: React.FC<Props> = ({onPress, styles, text}) => (
  <TouchableOpacity onPress={onPress} style={[styles?.boxStyle]} testID="box">
    <Text style={[styles?.textStyle]} testID="text">
      {text}
    </Text>
  </TouchableOpacity>
);

export default BaseButton;
