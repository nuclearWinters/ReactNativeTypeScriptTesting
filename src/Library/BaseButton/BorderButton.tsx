import React from 'react';
import BaseButton, {Props} from './BaseButton';
import {StyleSheet} from 'react-native';

const defaultStyles = StyleSheet.create({
  boxStyle: {flex: 1, alignItems: 'center', justifyContent: 'center'},
  textStyle: {
    fontSize: 18,
    borderWidth: 1,
    paddingHorizontal: 18,
    paddingVertical: 8,
    borderRadius: 6,
  },
});

const {boxStyle, textStyle} = defaultStyles;

const BorderButton: React.FC<Props> = ({onPress, text, styles}) => (
  <BaseButton
    styles={{
      boxStyle: [boxStyle, styles?.boxStyle],
      textStyle: [textStyle, styles?.textStyle],
    }}
    onPress={onPress}
    text={text}
  />
);

export default BorderButton;
