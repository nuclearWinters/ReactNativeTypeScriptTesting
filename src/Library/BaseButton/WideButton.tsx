import React from 'react';
import BaseButton, {Props} from './BaseButton';
import {StyleSheet} from 'react-native';

const defaultStyles = StyleSheet.create({
  boxStyle: {
    alignSelf: 'center',
    marginVertical: 64,
    backgroundColor: 'rgb(0,100,180)',
    width: '80%',
    paddingVertical: 12,
    borderRadius: 100,
  },
  textStyle: {
    fontSize: 24,
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
  },
});

const {boxStyle, textStyle} = defaultStyles;

const WideButton: React.FC<Props> = ({onPress, text, styles}) => (
  <BaseButton
    styles={{
      boxStyle: [boxStyle, styles?.boxStyle],
      textStyle: [textStyle, styles?.textStyle],
    }}
    onPress={onPress}
    text={text}
  />
);
export default WideButton;
