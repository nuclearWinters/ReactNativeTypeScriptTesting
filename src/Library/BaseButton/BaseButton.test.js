import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import BaseButton from './BaseButton';

test('BaseButton', async () => {
  const onPress = () => {};
  const text = 'Test';
  const {getByTestId, getByText, queryByTestId, baseElement} = render(
    <BaseButton onPress={onPress} text={text} />,
  );
  const button = getByTestId('box');
  fireEvent.press(button);
  expect(queryByTestId('box')).toBeTruthy();
  const textButton = getByTestId('text');
  expect(textButton.props.children).toBe(text);
  expect(baseElement).toMatchSnapshot();
});
