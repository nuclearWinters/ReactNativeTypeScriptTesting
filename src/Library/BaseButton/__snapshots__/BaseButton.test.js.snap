// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      style={
        Array [
          undefined,
        ]
      }
      testID="box"
    >
      <Text
        style={
          Array [
            undefined,
          ]
        }
        testID="text"
      >
        Test
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
